spec/models/patient_session/consent_spec.rb

```diff
@@ -0,0 +1,146 @@
+# frozen_string_literal: true
+
+describe PatientSession::Consent do
+  subject(:instance) { described_class.new(patient_session) }
+
+  let(:programme) { create(:programme, :hpv) }
+  let(:patient) { create(:patient, year_group: 8) }
+  let(:patient_session) do
+    create(:patient_session, programmes: [programme], patient:)
+  end
+
+  before { patient.strict_loading!(false) }
+
+  describe "#status" do
+    subject(:status) { instance.status.fetch(programme) }
+
+    context "with no consent" do
+      it { should be(described_class::NONE) }
+    end
+
+    context "with an invalidated consent" do
+      before { create(:consent, :invalidated, patient:, programme:) }
+
+      it { should be(described_class::NONE) }
+    end
+
+    context "with a not provided consent" do
+      before { create(:consent, :not_provided, patient:, programme:) }
+
+      it { should be(described_class::NONE) }
+    end
+
+    context "with both an invalidated and not provided consent" do
+      before do
+        create(:consent, :invalidated, patient:, programme:)
+        create(:consent, :not_provided, patient:, programme:)
+      end
+
+      it { should be(described_class::NONE) }
+    end
```

Comment:
I might be super paranoid here, but is it worth adding a spec for "refusal (invalidated) + given" here?

Reply:
Yep I think that's a good idea, I've added that in.